### PR TITLE
Add project report - pmt #99285

### PR DESCRIPTION
**Pull request by nikolas**
2015-04-08-124654_698x455_scrot

This commit adds postgresql-specific method - users_active_between()
to the Project model, and adds the result of that method to the
ProjectDetail context. This means that any test that accesses the
project detail page now needs a @skipUnless(postgresql) decorator.

I think this is okay, but if anyone disagrees I can look into
one of these alternatives:
- Implement users_active_between() in Django's ORM syntax. I couldn't
  think of a way to do this without making multiple queries, but maybe
  I'm missing something.
- Load the results of users_active_between() dynamically in JavaScript.

--

This commit also includes an update to bootstrap-datepicker, and a
tweak to how we're using jquery in requirejs.

I'm configuring the jQuery we're using with requirejs to not export
the global $ and jQuery variables, per these instructions here:
http://requirejs.org/docs/jquery.html#noconflictmap

Because I'm including bootstrap from outside of requirejs, bootstrap
was attaching itself to an inaccessible instance of jQuery.

The setup is now like this:
- A private jQuery, only accessible as an AMD module within requirejs.
- A global jQuery, which bootstrap attaches itself to

This necessitates wrapping the bootstrap-datepicker and typeahead
libraries in a define() call until these issues are resolved:
  https://github.com/twitter/typeahead.js/issues/922
  https://github.com/eternicode/bootstrap-datepicker/pull/1196

### CODE REVIEW

File: dmt/main/models.py [hunk 1]
```diff
@@ -710,6 +710,31 @@ def all_actual_times(self):
             item__milestone__project=self,
         ).order_by('completed')
 
+    def users_active_between(self, start, end):
```
Comment (sdreher) [inline dmt/main/models.py]: imho, the raw sql is ok if required for performance. The Django ORM aggregation functions are worth a look if you haven't already discounted them. https://docs.djangoproject.com/en/1.7/topics/db/aggregation/

File: dmt/main/tests/test_views.py [hunk 2]
```diff
@@ -134,63 +134,79 @@ def test_add_client(self):
         self.assertEqual(c.user, self.u)
 
 
-class TestProjectViews(TestCase):
+class TestProjectViews(LoggedInTestMixin, TestCase):
```
Comment (sdreher) [inline dmt/main/tests/test_views.py]: LoggedInTestMixin is cool. I hadn't seen that before.

**Comment by sdreher** [comment]
Let me know if you just want this merged. No worries about the raw SQL comment if you've already dismissed the ORM aggregation stuff.

**Comment by nikolas** [comment]
I'm looking at the ORM aggregation stuff now. I hadn't read this documentation before, and I think that what I want to do might be possible with the ORM. I'll give it a try and let you know.

File: dmt/main/models.py (hunk 1 above)
Comment (nikolas) [inline dmt/main/models.py]: I rewrote this function using django's db aggregation features. The function is actually simpler now.. thanks for pointing me in the right direction!

I removed all the test skips and they are all passing with a local `make jenkins`, so this PR is ready for further review/merge.

**Pull request merged**